### DISCUSSION TITLE
For a specific dependency how can I get it to only update on major versions? Whilst leaving other updates as normal

### DISCUSSION THREAD

**Discussion by paddyroddy** [renovatebot/renovate#25578] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

My (relevant bits of) config is
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:best-practices",
	"schedule:weekdays",
    ":automergeBranch",
    ":automergeDigest",
    ":automergeMinor",
    ":automergePatch",
    ":enablePreCommit"
  ]
}
```
so essentially I want to automerge for all non-major options. I want to verify my config is valid using https://github.com/renovatebot/pre-commit-hooks. However, as this is a `pre-commit` manager, it is updating in the usual way. And it updates very often.

I was wondering if it was possible `"matchDepNames": ["renovatebot/pre-commit-hooks"]` to update only on `major` versions?

### What you tried so far.

For now to reduce the updates I've changed the schedule
```json
"packageRules": [
  {
    "description": "Only update once a month",
    "matchDepNames": ["renovatebot/pre-commit-hooks"],
    "schedule": ["on the first day of the month"]
  }
]
```
which I think is valid config?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by secustor** [comment, marked as answer]
```json
"packageRules": [
  {
    "description": "Only allow major standard upgrades",
    "matchDepNames": ["renovatebot/pre-commit-hooks"],
    "matchUpdateTypes": ["minor", "patch", "pin"],
    "enabled": false
  }
]
```
https://docs.renovatebot.com/configuration-options/#matchupdatetypes

**Reply by paddyroddy** [reply]
Ah great idea, thanks!

**Reply by HonkingGoose** [reply]
I recommend you use Dependency Dashboard approval for all but the `major` updates to `renovatebot/pre-commit-hooks`. That way you'll always _see_ the rest of the updates for the package in the Dependency Dashboard. With `@secustor`'s config, you're essentially making any other updates to the package _invisible_.

```json
{
  "packageRules": [
    {
      "description": "Require Dependency Dashboard approval for all updates to renovatebot/pre-commit-hooks, except for major updates.",
      "matchDepNames": ["renovatebot/pre-commit-hooks"],
      "matchUpdateTypes": ["minor", "patch", "pin", "pinDigest", "digest", "lockFileMaintenance", "rollback", "bump", "replacement"],
      "dependencyDashboardApproval": true
    }
  ]
}
```

[Renovate docs, require Dependency Dashboard approval for specific packages](https://docs.renovatebot.com/key-concepts/dashboard/#require-approval-for-specific-packages)

---

Edit: I misread the topic a bit (I thought you wanted to ignore `major` updates...), so I've updated the config and text to match what you actually want to do... 😄

**Reply by paddyroddy** [reply]
Thanks, @HonkingGoose, do you know if there is a way of saying all `matchUpdateTypes` except `major`?

**Reply by HonkingGoose** [reply]
I don't know a easy way to say "all `matchUpdateTypes` except `major`" right now. I don't know if a creating a `matchAllUpdateTypesExceptMajor` preset is a good idea... 😄 

## `extends` into a `packageRules` object?

Do we want to support putting an `extends` into a `packageRules` object? Silly JSON incoming:

```json
{
  "extends": ["config:best-practices"],
  "packageRules": [
    {
      "description": "Require Dependency Dashboard approval for all updates to renovatebot/pre-commit-hooks, except for major updates.",
      "extends": ["matchAllUpdateTypesExceptMajor"],
      "dependencyDashboardApproval": true
    }
  ]
}
```

### `matchAllUpdateTypesExceptMajor` preset code:

```json
{
  "packageRules": [
    {
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "pinDigest",
        "digest",
        "lockFileMaintenance",
        "rollback",
        "bump",
        "replacement"
      ]
    }
  ]
}

```

**Reply by paddyroddy** [reply]
Haha, I kind of like it

**Reply by HonkingGoose** [reply]
The _real_ question is, will the Renovate maintainers _like_ this silly thing? 😉

**Reply by paddyroddy** [reply]
I gave `dependencyDashboardApproval` a go, but I really dislike the permanently open issue (I use `":disableDependencyDashboard"` 🤫), was hoping I could trigger it from https://developer.mend.io. So for now I've gone for @secustor's suggestion of `"enabled": false`

**Reply by xenoterracide** [reply]
trying to not match anything but major versions for gradle EXCEPT one settings plugin because those can't use dynamic versions with lockfiles. I've tried numerous variations, this is my latest. It seems to have worked for just getting major versions updated, but I can't get the develocity plugin to update

```json5
    {
      matchPackageNames: ["com.gradle.develocity"],
      matchManagers: ["gradle"],
      automerge: true,
    },
    {
      description: "Only allow major standard upgrades",
      matchManagers: ["gradle", "gradle-wrapper", "npm"],
      matchUpdateTypes: ["minor", "patch", "pin"],
      enabled: false,
    },
```
